### PR TITLE
improvement(eslint-config-fluid): Update type import/export rules to help prevent runtime side-effects

### PR DESCRIPTION
**Pull request by Josmithr**
Adds the [@typescript-eslint/no-import-type-side-effects](https://typescript-eslint.io/rules/no-import-type-side-effects/) rule to our strict eslint configuration.

Pre-fixes violations across the repo.

### CODE REVIEW

File: common/build/eslint-config-fluid/strict.js [hunk 1]
```diff
@@ -73,21 +73,32 @@ module.exports = {
...
+				 *
+				 * @see {@link https://typescript-eslint.io/rules/consistent-type-imports/}
 				 */
 				"@typescript-eslint/consistent-type-imports": [
 					"error",
 					{ fixStyle: "separate-type-imports" },
 				],
 
+				/**
+				 * Ensures that import statements containing only types do not result in runtime side-effects.
```
Comment (Josmithr) [inline common/build/eslint-config-fluid/strict.js:96]:
```suggestion
				 * Ensures that type-only import statements do not result in runtime side-effects.
```

**Review by noencke — APPROVED** [review]

File: common/build/eslint-config-fluid/strict.js [hunk 2]
```diff
@@ -73,21 +73,32 @@ module.exports = {
 				/**
 				 * Requires that type-only exports be done using `export type`. Being explicit allows the TypeScript
 				 * `isolatedModules` flag to be used, and isolated modules are needed to adopt modern build tools like swc.
+				 *
+				 * @see {@link https://typescript-eslint.io/rules/consistent-type-exports/}
 				 */
 				"@typescript-eslint/consistent-type-exports": [
 					"error",
-					{ fixMixedExportsWithInlineTypeSpecifier: false },
+					{ fixMixedExportsWithInlineTypeSpecifier: true },
```
Comment (tylerbutler) [inline common/build/eslint-config-fluid/strict.js:81]: Don't we prefer type-only blocks to inline type? Regardless this seems to be an unrelated change.

Comment (Josmithr) [reply common/build/eslint-config-fluid/strict.js:81]: Yeah, fair that it's unrelated. I'm personally not a fan of this pattern. I prefer a single statement for all exports from a single module, as it makes the export modules much easier to read IMO. But you're right that it doesn't belong in this PR.
```suggestion
					{ fixMixedExportsWithInlineTypeSpecifier: false },
```

File: common/build/eslint-config-fluid/strict.js [hunk 3]
```diff
@@ -82,12 +84,21 @@ module.exports = {
...
 					"error",
 					{ fixStyle: "separate-type-imports" },
 				],
 
+				/**
+				 * Ensures that type-only import statements do not result in runtime side-effects.
+				 *
+				 * @see {@link https://typescript-eslint.io/rules/no-import-type-side-effects/}
+				 */
+				"@typescript-eslint/no-import-type-side-effects": "error",
```
Comment (Josmithr) [inline common/build/eslint-config-fluid/strict.js:100]: For reviewers: this is the crux of the change

**Review by alexvy86 — APPROVED** [review]
Devtools changes look good
